cloud_utils/slack_utils.py

```diff
@@ -35,3 +46,85 @@ async def report_exception(webhook_url: str, text: str):
         _make_exception_payload,
         gamla.post_json_async(5, webhook_url),
     )
+
+
+async def post_to_notifications_and_dm_user(
+    message: str, email: str = "", email_map: Dict[str, str] = frozendict()
+) -> bool:
+    """
+    DM the email with the message, if given, and catch and log the exception if the email isn't right.
+    Then post the message to `#notifications`.
+    :return: a `dm_succeeded` boolean --> IFF the DM went through
+    """
+    dm_succeeded = False
+    if email:
+        dm_succeeded = dm_slack_user_and_log_error(email, message, email_map)
+        message += f" ({email})"
+    await post_to_notifications(message)
+    return dm_succeeded
```

Comment:
Consider the following instead.
It will allow you to delete `dm_slack_user_and_log_error` and simplify the code
```py
    try:
        dm_slack_user(email, message, email_map)
    except slack_sdk.errors.SlackApiError as e:
       logging.warning(f"Could not send DM to Slack email '{email}'. See error: {e}")
    await post_to_notifications(message)
```

Reply:
ok